scheduler.go

```diff
@@ -386,14 +387,6 @@ func in(scheduleWeekdays []time.Weekday, weekday time.Weekday) bool {
 }
 
 func (s *Scheduler) calculateDuration(job *Job) time.Duration {
-	if job.neverRan() && shouldRunAtSpecificTime(job) { // ugly. in order to avoid this we could prohibit setting .At() and allowing only .StartAt() when dealing with Duration types
```

Comment:
I think this [check](https://github.com/go-co-op/gocron/blob/fb8a25590a070515d6069a3dbb31a66e40c3ebdd/scheduler.go#L853) makes this dead code since `shouldRunAtSpecificTime` will never be true (`job.getAtTime(jobLastRun) != 0` and under hour unit should be already flagged by an error)